src/Sushi.Mediakiwi.API/Extensions/Extensions.cs

```diff
@@ -0,0 +1,144 @@
+﻿using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Primitives;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.OpenApi.Models;
+using Sushi.Mediakiwi.API.Services;
+using Swashbuckle.Swagger;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Sushi.Mediakiwi.API.Extensions
+{
+    public static class Extensions
+    {
+
+        public static HttpContext Clone(this HttpContext httpContext, bool copyBody)
+        {
+            var existingRequestFeature = httpContext.Features.Get<IHttpRequestFeature>();
+
+            var requestHeaders = new Dictionary<string, StringValues>(existingRequestFeature.Headers.Count, StringComparer.OrdinalIgnoreCase);
+            foreach (var header in existingRequestFeature.Headers)
+            {
+                requestHeaders[header.Key] = header.Value;
+            }
+
+            var requestFeature = new HttpRequestFeature
+            {
+                Protocol = existingRequestFeature.Protocol,
+                Method = existingRequestFeature.Method,
+                Scheme = existingRequestFeature.Scheme,
+                Path = existingRequestFeature.Path,
+                PathBase = existingRequestFeature.PathBase,
+                QueryString = existingRequestFeature.QueryString,
+                RawTarget = existingRequestFeature.RawTarget,
+                Headers = new HeaderDictionary(requestHeaders),
+            };
+
+            if (copyBody)
+            {
+                // We need to buffer first, otherwise the body won't be copied
+                // Won't work if the body stream was accessed already without calling EnableBuffering() first or without leaveOpen
+                httpContext.Request.EnableBuffering();
```

Comment:
this is a performance killer, don't do this

why do we need to clone the httpcontext?